src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Providers/Commands.RecoveryServices.Backup.Providers.csproj

```diff
@@ -88,6 +88,10 @@
       <Project>{b758fec1-35c1-4f93-a954-66dd33f6e0ec}</Project>
       <Name>Commands.RecoveryServices.Backup.ServiceClientAdapter</Name>
     </ProjectReference>
+    <ProjectReference Include="..\Commands.RecoveryServices\Commands.RecoveryServices.csproj">
```

Comment:
Why do you need to add this reference?

Reply:
This project contains the definition of ARSVault which is being referenced by the methods in the provider files.

Reply:
@dragonfly91 until all of the RecoveryServices modules are consolidated into one, we should avoid introducing any coupling between the types -- this prevents us from getting into scenarios where users can only use certain Backup functionality when paired with RecoveryServices cmdlets. Once the modules are consolidated, we can have a common models projects that all of the projects share and can use interchangeably without worrying about having additional modules installed and run beforehand.

Reply:
@dragonfly91 as a workaround, we could have the user provide the vault resource id, which you can parse using the [`ResourceIdentifier`](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Common/Commands.ResourceManager.Common/Utilities/Models/ResourceIdentifier.cs) class, to get the vault name and resource group, which appear to be the only two properties you are using from the vault object (let me know if that's not the case)

Reply:
@cormacpayne 

When we had this discussion last time regarding merge, this point about how to address module dependency was never mentioned explicitly by the ADX team, I believe. Removing dependency on the Set-AzureRmRecoveryServicesVaultContext cmdlet is one of the biggest pain points for the customers and providing the Vault object is the most comfortable way for them. @pvrk can you comment on the workaround suggested by @cormacpayne ?

Also, this PR has been open for the last 9 days. This comment, given on the 9th day, essentially calls for a complete code revamp; all the files will have to be modified. 

I understand your commitment to make things easy for the customer. But, I feel that there is no way to anticipate the PR review process. This apparent issue in our design could have been:

1. Educated in the initial talks about merge
2. Flagged in the PR Design Review
3. Flagged in the first few days of PR Code Review.

My concern: why didn't this comment come up in any of the above steps? We have now worked on throw away code and also risk missing the release. 

Hope you understand our pain point and we need you to address this.

Reply:
@cormacpayne : So the idea is to use vault ID as a workaround and switch it back to object when the code merger is done? Even though if the idea is to use vault ID throughout, it came at the very last minute at a very high cost (we need to modify all the files) and I concur with @dragonfly91 here. We need such inputs as early as possible.